kyo-caliban/src/main/scala/kyo/resolvers.scala

```diff
@@ -0,0 +1,65 @@
+package kyo
+
+import caliban.*
+import caliban.interop.tapir.*
+import caliban.interop.tapir.TapirAdapter.*
+import kyo.internal.KyoSttpMonad
+import sttp.tapir.Codec.JsonCodec
+import sttp.tapir.Endpoint
+import sttp.tapir.capabilities.NoStreams
+import sttp.tapir.server.ServerEndpoint
+import sttp.tapir.server.netty.*
+import zio.*
+import zio.stream.ZStream
+
+opaque type Resolvers = Aborts[Throwable] & ZIOs
+
+object Resolvers:
+
+    private given StreamConstructor[Nothing] =
+        (_: ZStream[Any, Throwable, Byte]) => throw new Throwable("Streaming is not supported")
+
+    def run[T, S](v: HttpInterpreter[Any, CalibanError] < (Resolvers & S)): NettyKyoServerBinding < (ZIOs & Aborts[Throwable] & S) =
+        run[T, S](NettyKyoServer())(v)
+
+    def run[T, S](server: NettyKyoServer)(v: HttpInterpreter[Any, CalibanError] < (Resolvers & S))
+        : NettyKyoServerBinding < (ZIOs & Aborts[Throwable] & S) =
+        ZIOs.get(ZIO.runtime[Any]).map(runtime => run(server, runtime)(v))
+
+    def run[R, T, S](runner: Runner[R])(v: HttpInterpreter[Runner[R], CalibanError] < (Resolvers & S))(using
+        tag: Tag[Runner[R]]
+    ): NettyKyoServerBinding < (ZIOs & Aborts[Throwable] & S) =
+        run[R, T, S](NettyKyoServer(), runner)(v)
+
+    def run[R, T, S](server: NettyKyoServer, runner: Runner[R])(v: HttpInterpreter[Runner[R], CalibanError] < (Resolvers & S))(using
+        tag: Tag[Runner[R]]
+    ): NettyKyoServerBinding < (ZIOs & Aborts[Throwable] & S) =
+        ZIOs.get(ZIO.runtime[Any]).flatMap(runtime => run(server, runtime.withEnvironment(ZEnvironment(runner)))(v))
+
+    def run[R, T, S](
+        server: NettyKyoServer,
+        runtime: Runtime[R]
+    )(v: HttpInterpreter[R, CalibanError] < (Resolvers & S)): NettyKyoServerBinding < (ZIOs & Aborts[Throwable] & S) =
+        for
+            interpreter <- v
+            endpoints = interpreter.serverEndpoints[R, NoStreams](NoStreams).map(convertEndpoint(_, runtime))
+            bindings <- IOs(server.addEndpoints(endpoints).start())
+        yield bindings
+
+    def get[R](api: GraphQL[R])(using
+        requestCodec: JsonCodec[GraphQLRequest],
+        responseValueCodec: JsonCodec[ResponseValue]
+    ): HttpInterpreter[R, CalibanError] < Resolvers =
+        ZIOs.get(api.interpreter.map(HttpInterpreter(_)))
+
+    private def convertEndpoint[R, I](
+        endpoint: ServerEndpoint.Full[Unit, Unit, I, TapirResponse, CalibanResponse[NoStreams.BinaryStream], NoStreams, [x] =>> RIO[R, x]],
+        runtime: Runtime[R]
+    ): ServerEndpoint[Any, KyoSttpMonad.M] =
+        ServerEndpoint[Unit, Unit, I, TapirResponse, CalibanResponse[NoStreams.BinaryStream], Any, KyoSttpMonad.M](
+            endpoint.endpoint.asInstanceOf[Endpoint[Unit, I, TapirResponse, CalibanResponse[NoStreams.BinaryStream], Any]],
+            _ => _ => Right(()),
+            _ => _ => req => Unsafe.unsafe { implicit u => runtime.unsafe.run(endpoint.logic(zioMonadError)(())(req)).getOrThrow() }
```

Comment:
I'm taking a look if we could avoid unsafe here by allowing a type parameter `S` in `NettyKyoServer`.

Reply:
I wanted to allow `ZIOs` with `NettyKyoServer` but it's not trivial because `MonadAsyncError` requires `IOs.ensure` that can't accept arbitrary effects. I think we could eventually integrate ZIO's `Scope` with Kyo's `Resources` to address that, though. What are the implications of using unsafe here? I imagine it'd use thread blocking and not propagate cancellations/interrupts?

Reply:
I agree it's not great since it will just block. I will see what I can do with other `run` variants such as `runToFuture` or `fork` 🤔

Reply:
I change the implementation to re-use `KyoSttpMonad.async` with `runToFuture`. Let me know what you think 🙏

Reply:
Nice! I didn't know ZIO had a `Future` subclass with cancellation. We should have something similar.